siriuspy/siriuspy/devices/fofb_acq.py

```diff
@@ -894,6 +970,30 @@ def check_prbs_bpmposy_levels(self, level0, level1=None):
                 return False
         return True
 
+    def wait_prbs_bpmposy_levels(
+            self, level0, level1=None, timeout=_FamFOFBAcqBase.DEF_TIMEOUT):
+        """
+        Wait for BPM PosY PRBS excitation levels to be equal to
+        desired values.
+
+        Args:
+            level0 (numpy.ndarray, (NR_BPM,)):
+                desired values for BPM PosY PRBS excitation level 0.
+            level1 (optional, numpy.ndarray, (NR_BPM,)):
+                If None, we consider level1 = -level0. Defaults to None.
+
+        Returns:
+            bool: indicate whether levels 0 and 1 of BPM PosY PRBS
+                excitation are in desired values.
+
+        """
+        _t0 = _time.time()
+        while _time.time() - _t0 < timeout:
+            if self.check_prbs_bpmposy_levels(level0, level1):
+                return True
+            _time.sleep(0.1)
+        return False
```

Comment:
I wonder if siriuspy couldn't use a `wait_timeout` function which takes a "check function" and a timeout as parameter and implements this same loop (maybe have the sleep time as a default value).

Reply:
Agreed! I will do this in a future PR, with other improvements in the same direction (as we talked offline)